File: 0xxx/00xx/002x/0020.ts
function isValid(s: string): boolean {
  if (s.length % 2) return false

  const map: { [key: string]: number } = {
    '(': 1,
    '[': 2,
    '{': 3,
    ')': 1,
    ']': 2,
    '}': 3
  }
  const recorder: number[] = []

  for (let i = 0; i < s.length; i++) {
    if (s[i] === '(' || s[i] === '[' || s[i] === '{') {
      recorder.push(map[s[i]])
    } else if (map[s[i]] !== recorder.pop()) {
      return false
    }
  }
  return !recorder.length
}

File: 0xxx/00xx/002x/0020.py
class Solution:
  def isValid(self, s: str) -> bool:
    if len(s) % 2 != 0: return False

    bracketMap = {'(': 1, '[': 2, '{': 3, ')': 1, ']': 2, '}': 3}
    recorder = []

    for char in s:
      if char in '([{':
        recorder.append(bracketMap[char])
      elif len(recorder) == 0:
        return False
      elif recorder.pop() != bracketMap[char]:
        return False
    return len(recorder) == 0